### PR TITLE
Add Libp2p Noise security protocol support

### PR DESCRIPTION
**Pull request by Nashatyrev**
## PR Description

- bump jvm-libp2p version to `0.3.3` which has fixed Noise issues (see https://github.com/libp2p/jvm-libp2p/pull/99 for details)
- Add `noise` as a primary security protocol 

TBD: remove `secio` protocol when all clients support compatible `noise` versions

### CODE REVIEW

File: networking/p2p/src/main/java/tech/pegasys/artemis/networking/p2p/libp2p/LibP2PNetwork.java [hunk 1]
```diff
@@ -116,6 +117,7 @@ public LibP2PNetwork(
             b -> {
               b.getIdentity().setFactory(() -> privKey);
               b.getTransports().add(TcpTransport::new);
+              b.getSecureChannels().add(NoiseXXSecureChannel::new);
```
Comment (cemozerr) [inline networking/p2p/src/main/java/tech/pegasys/artemis/networking/p2p/libp2p/LibP2PNetwork.java:120]: By adding Noise above Secio, do we preferentially use Noise now?

**Review by cemozerr — APPROVED** [review]
LGTM. Asked a question about preferentially using Noise

File: networking/p2p/src/main/java/tech/pegasys/artemis/networking/p2p/libp2p/LibP2PNetwork.java (hunk 1 above)
Comment (Nashatyrev) [reply networking/p2p/src/main/java/tech/pegasys/artemis/networking/p2p/libp2p/LibP2PNetwork.java:120]: Yes. When initiating connection we would try to negotiate `noise` first
No more preferences besides this. `secio` would also be supported in both initiator and responder roles

**Review by mbaxter — APPROVED** [review]
nice!! 🎉

**Pull request merged**